src/condor_dagman/dagman_submit.cpp

```diff
@@ -541,11 +541,9 @@ static void init_dag_vars(SubmitHash * submitHash,
 	}
 
 	//PRAGMA_REMIND("TODO: fix the tests to use $(DAG_PARENT_NAMES), and then remove custom job attribute")
-	if (!parents.empty()) {
```

Comment:
why is it no longer possible that parents() be empty?

Reply:
It is possible that parents() be empty. However the `$$([DAGParentNodeNames])` expansion assumes that it will always find a `DAGParentNodeNames` attribute in the classad, even when no parents exist. So we always want to publish this attribute, even if empty.